pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -154,6 +156,14 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		return err
 	}
 
+	if reasons := maintainFeatureGatesForShoot(maintainedShoot); len(reasons) > 0 {
+		operations = append(operations, reasons...)
+	}
+
+	if reasons := maintainAdmissionPlugins(maintainedShoot); len(reasons) > 0 {
```

Comment:
Cosmetic nit:

You could add `ForShoot` suffix there 
```suggestion
	if reasons := maintainAdmissionPluginsForShoot(maintainedShoot); len(reasons) > 0 {
```

or drop the `ForShoot` suffix from `maintainFeatureGatesForShoot`. Just for consistency reasons that both funcs have similar names.